src/components/Footer.tsx

```diff
@@ -0,0 +1,89 @@
+import { Dispatch, FC } from 'react';
+
+import { Status } from '../types/Status';
+import { Todo } from '../types/Todo';
+import { deleteTodos } from '../api/todos';
+
+interface Props {
+  todos: Todo[];
+  selectedStatus: Status;
+  setSelectedStatus: (status: Status) => void;
+  setTodos: Dispatch<React.SetStateAction<Todo[]>>;
+  setIsDeleting: Dispatch<React.SetStateAction<number>>;
+  onErrorMessage: (message: string) => void;
+}
+
+const Footer: FC<Props> = ({
+  todos,
+  selectedStatus,
+  setSelectedStatus,
+  setTodos,
+  setIsDeleting,
+  onErrorMessage,
+}) => {
+  const activeTodosCount = todos.filter(todo => !todo.completed).length;
+  const completedTodos = todos.filter(todo => todo.completed);
+
+  const handleClearCompleted = () => {
+    completedTodos.forEach(todo => {
+      setIsDeleting(todo.id);
+      deleteTodos(todo.id)
+        .then(() =>
+          setTodos((prevState: Todo[]) =>
+            prevState.filter(t => t.id !== todo.id),
```

Comment:
Using one letter in the callback ( or other) function is not a good idea. Please fix it in future iterations.